script.sh

```diff
@@ -177,12 +178,28 @@ fi
 
 cat >>Dockerfile <<EOF
 RUN apt-get update && apt-get dist-upgrade --yes
-RUN apt-get install --yes --no-install-recommends build-essential equivs devscripts git-buildpackage
+RUN apt-get install --yes --no-install-recommends build-essential equivs devscripts git-buildpackage apt-transport-https wget ca-certificates
 
 WORKDIR $(pwd)
 COPY . .
 EOF
 
+if [ -n "${TRAVIS_EXTRA_REPOSITORY_GPG_URL}" ]
+then
+	cat >>Dockerfile <<EOF
+RUN wget -O- "${TRAVIS_EXTRA_REPOSITORY_GPG_URL}" | apt-key add -
+EOF
+fi
+
+if [ -n "${TRAVIS_EXTRA_REPOSITORY}" ]
+then
+	cat >>Dockerfile <<EOF
+RUN echo "deb ${TRAVIS_EXTRA_REPOSITORY}" >> /etc/apt/sources.list
+RUN echo "deb-src ${TRAVIS_EXTRA_REPOSITORY}" >> /etc/apt/sources.list
+RUN apt-get update
```

Comment:
> unless you really want me to split it.

Into some kind of "extra_packages" or similar? Not sure we should be just throwing apt-transport-https in there by default unless for good reason..?